tests/test_file_utils.py

```
@@ -0,0 +1,1078 @@
+        with tempfile.TemporaryDirectory() as temp_dir:
+            file_path = os.path.join(temp_dir, "subdir", "test.json")
+            data = {"test": "data"}
+
+            success = write_json_file(file_path, data)
+            assert success == True
+            assert os.path.exists(file_path)
+
+    def test_write_json_file_invalid_data(self):
+        """Test writing invalid JSON data"""
+        # Create data that can't be serialized
+        invalid_data = {"key": set([1, 2, 3])}  # sets are not JSON serializable
```

Ruff (0.11.9)

Unnecessary list literal (rewrite as a set literal)

Rewrite as a set literal

(C405)